src/expr/src/linear.rs

```diff
@@ -1881,6 +1882,132 @@ pub mod plan {
+            // TODO: This is entirely just a placeholder impl so that we can get
+            // at least one query to work end-to-end. The real structure would
+            // certainly look different than this.
+            //
+            // It works by sniffing out expressions that (at a high level) look
+            // like `some_col {<,<=,>=,>} constant`. For `<`, if we substitute
+            // the minimal value in the part and if `some_col_min < constant` is
+            // _false_, then we know nothing else in the part would be true and
+            // we can skip fetching it. Analogous logic holds for the other
+            // comparisons, but with the max substituted for `{>=,>}`.
+            //
+            // The full generalization probably requires being able to reason
+            // about the range of outputs of functions given a restricted input
+            // domain, but... that seems hard? There's probably some nice middle
+            // ground.
+            if let MirScalarExpr::CallBinary { func, expr1, expr2 } = expr {
+                let col_idx = Self::one_column_reference(expr1)?;
+                // WIP: Should the `col_min` and `col_max` methods instead take
+                // indexes? That would allow us to remove the RelationDesc from
+                // MfpPushdown.
+                let col_name = self.relation_desc.get_name(col_idx);
+                let min_or_max = match func {
+                    BinaryFunc::Lt | BinaryFunc::Lte => stats.col_min(col_name.as_str(), &arena)?,
+                    BinaryFunc::Gt | BinaryFunc::Gte => stats.col_max(col_name.as_str(), &arena)?,
+                    _ => return None,
+                };
+                if !Self::zero_column_references(expr2)? {
+                    return None;
+                }
+                let mut datums = Vec::with_capacity(self.relation_desc.arity());
+                while datums.len() < col_idx {
+                    datums.push(Datum::Null);
+                }
+                datums.push(min_or_max);
+                while datums.len() < self.relation_desc.arity() {
+                    datums.push(Datum::Null);
+                }
+                match expr.eval(datums.as_slice(), &arena) {
+                    Ok(Datum::False) => return Some(true),
+                    _ => return Some(false),
+                }
+            }
+            None
+        }
+
+        fn zero_column_references(expr: &MirScalarExpr) -> Option<bool> {
+            match expr {
+                MirScalarExpr::Literal(..) => Some(true),
+                _ => None,
+            }
+        }
+
+        fn one_column_reference(expr: &MirScalarExpr) -> Option<usize> {
+            match expr {
+                MirScalarExpr::Column(idx) => Some(*idx),
+                MirScalarExpr::CallUnary {
+                    func:
+                        UnaryFunc::CastUint64ToNumeric(..) | UnaryFunc::CastUint64ToMzTimestamp(..),
```

Comment:
`invert_casts_on_expr_eq_literal_inner` has a similar logic for equalities instead of inequalities. It relies on the `inverse` and `preserves_uniqueness` function annotations, [so it works for all those sneaky implicit casts](https://github.com/MaterializeInc/materialize/issues/15525), and even for some other unary functions. 

Unfortunately, it can't be applied here out of the box, because some inverse functions might flip the `>` to `<`. For example, we have an inverse for integer negation (itself), and a flip happens when we apply it. But maybe we could adapt it: we could introduce an additional function annotation besides `preserves_uniqueness` to mark an inverse function safe for applying on both sides of an inequality.